README.md

```diff
@@ -252,12 +268,58 @@ If the payment instrument specified by `instrumentId` is not available or if the
 
 🚨 As a hack for the [pilot], the user agent will simply resolve `request.show()` with an exception. The caller is responsible for constructing a second Payment Request to open `fallbackUrl` inside a Secure Modal Window by abusing the Just-in-Time registration and skip-the-sheet features of Payment Handler API.
 
+### [Future] Using an RP provided Payment Handler
+
+The caller can invoke [Payment Request API] before the user selects a payment instrument and still leverage Secure Payment Confirmation. In this case the browser will prompt the user to select a payment instrument and then invoke the RP's Payment Handler which will invoke Secure Payment Confirmation.
+
+> **NOTE:** There is not a great developer experience when using SPC and Payment Handlers as yet so the correlation is hacky. During the installation flow for a Payment Handler it is possible for the RP to [store](https://w3c.github.io/payment-handler/#set-method) the details of one or more payment instruments linked to the Payment Handler. The following behaviour is backwards compatible with existing [Payment Handler API] behaviour but could be improved in future.
+
+ 1. For each of the `supportedMethods` provided in the call to [Payment Request API] the browser will gather the set of Payment Handlers that support the given method.
+ 2. For each of the instruments linked to a Payment Handler in the returned set the browser will extract the `instrumentKey` provided when adding the instrument to the Payment Handler and look for a matching `instrumentId` among all of the instruments linked to a PaymentCredential. If it finds a match the instrument is added to the set of user choices.
+ 3. If the set of user choices is not empty the user is prompted to select a payment instrument, else the browser reverts to the usual Payment Handler selection flow.
+ 4. If the user selects an instrument the corresponding Payment Handler is invoked and the `instrumentId` is passed to it as the [`instrumentKey`](https://w3c.github.io/payment-handler/#dom-paymentrequestevent-instrumentkey) property of the `PaymentRequestEvent`.
+ 
+ If the user's chosen instrument can be used for Secure Payment Confirmation (i.e. is linked to a credential) `PaymentRequestEvent.isSecureConfirmationAvailable` will be `true`.
+ 
+ The Payment Handler can can query its backend systems to get the neccessary challenge (`challenge`) and verify that the `instrumentId` is still valid before calling `PaymentRequestEvent.securePaymentConfirmation(challenge, timeout)` to invoke the Secure Payment Confirmation flow.
+ 
+The browser will display a native user interface with the payment amount and the payee origin, which is taken to be the origin of the top-level context where the `PaymentRequest` API was invoked.
```

Comment:
Can we move this step before triggering the `PaymentRequestEvent`? I think functionally the flow will be identical.

To mitigate the privacy threat of sharing data to a 3rd party before user awareness, I'd like to make sure all new flows we design are compatible with the "native UI first" principle. Browser will only release data to the payment handler if the user is aware that they're about to start a payment by clicking "Verify" on the native browser UI.

Reply:
A few issues/comments:

 1. It's going to be a problem in future if the UI is combined and the payment details are rendered by the authenticator as we'll already need the challenge from the RP.
 2. Assuming the user has already selected an instrument they have already been presented with a native UI and implicitly chosen to interact with the issuer of that instrument.
 3. If the instrument data is stale the user may have chosen an instrument that subsequently gets rejected by the RP when the PH is invoked and goes online to get the challenge.